main/rfc1867.c

```diff
@@ -1138,20 +1135,42 @@ SAPI_API SAPI_POST_HANDLER_FUNC(rfc1867_post_handler) /* {{{ */
 				s = filename;
 			}
 
-			if (!is_anonymous) {
-				safe_php_register_variable(lbuf, s, strlen(s), NULL, 0);
-			}
-
 			/* Add $foo[name] */
 			if (is_arr_upload) {
 				snprintf(lbuf, llen, "%s[name][%s]", abuf, array_index);
 			} else {
 				snprintf(lbuf, llen, "%s[name]", param);
 			}
 			register_http_post_files_variable(lbuf, s, &PG(http_globals)[TRACK_VARS_FILES], 0);
-			efree(filename);
 			s = NULL;
 
+			/* Add $foo_fullpath */
+			if (llen < strlen(param) + MAX_SIZE_OF_INDEX + 1) {
+				llen = (int)strlen(param);
+				lbuf = (char *) safe_erealloc(lbuf, llen, 1, MAX_SIZE_OF_INDEX + 1);
+				llen += MAX_SIZE_OF_INDEX + 1;
+			}
```

Comment:
This code doesn't need to be repeated. You might have needed to adjust the value of MAX_SIZE_OF_INDEX, but it's currently `sizeof("[tmp_name]")`, which is the same as `sizeof("[fullpath]")`. Would need an adjustment if the name if changed though (to `[full_path]` for example).